extensions/worker-specific-hashrate-tracking.md

```diff
@@ -0,0 +1,105 @@
+
+### 1.1 Activate Extension
+
+A client wishing to use this extension sends the `Activate` message. If supported, the pool responds with `Activate.Success`. Otherwise, the client stops further attempts.
+
+---
+
+## 2. Extension Messages
+
+### `Activate` (Client -> Server)
+
+| Field Name | Data Type | Description                                                 |
+| ---------- | --------- | ----------------------------------------------------------- |
+| request_id | U32       | Unique identifier for pairing the response                  |
+
+### `Activate.Success` (Server -> Client)
+
+| Field Name | Data Type | Description                                                 |
+| ---------- | --------- | ----------------------------------------------------------- |
+| request_id | U32       | Unique identifier for pairing the response                  |
+
+### `SetUserIdentity` (Client -> Server)
+
+The client synchronizes the `user_id` ↔ `user_identity` mapping with the server (Pool).
+
+| Field Name    | Data Type        | Description                           |
+|---------------|------------------|---------------------------------------|
+| channel_id    | U32              | Channel identifier                    |
+| request_id    | U32              | Unique request ID                     |
+| user_id       | SEQ0_64K[U32]    | Unique IDs for each worker            |
+| user_identity | SEQ0_64K[STR0_255]| Worker identities (names)            |
+
+### `SetUserIdentity.Success` (Server -> Client)
+
+Response indicating the pool has accepted the mapping.
+
+| Field Name    | Data Type    | Description                               |
+|---------------|--------------|-------------------------------------------|
+| channel_id    | U32          | Channel ID                                |
+| request_id    | U32          | Original request ID                       |
+| user_id       | SEQ0_64K[U32]| Worker ID associated                      |
+
+### `SetUserIdentity.Error` (Server -> Client)
+
+Sent if there’s an error with the `user_identity` mapping configuration.
+
+| Field Name    | Data Type   | Description                                |
+|---------------|-------------|--------------------------------------------|
+| channel_id    | U32         | Channel ID                                 |
+| request_id    | U32         | Original request ID                        |
+| error_code    | STR0_255    | Error reason                               |
+
+Possible error codes:
+- `invalid-channel-id`
+
+### Two different options for Submitting Shares with `user_id`
+
+This extension offers two possible approaches for submitting shares with `user_id`, allowing for implementation via one of the following methods:
```

Comment:
I don't understand why there are two options here? They are the same data.

Reply:
I put two options there because I would like to discuss the best option to push with as many of you as possible. @Fi3 for example was suggesting to create a freshly new message to submit shares with id (which is the new `SubmitIdentifiedSharesExtended` message). The alternative approach I wanted to add is about simply expanding the current `SubmitSharesExtended` message with the new `user_id` field.
We need to decide which is the best approach between these two. Does it make sense?

Reply:
I prefer making this new functionality explicit on the message name, ortherwise having multiple conflicting versions of `SubmitSharesExtended` could cause confusion

+1 for `SubmitIdentifiedSharesExtended`

Reply:
I think it can be a good solution to have separared SubmitIdentifiedSharesExtended

Reply:
Removed option 1 and left `SubmitIdentifiedSharesExtended` in 27d1751ed9371b90b9c5e8d0c4cf0960920503ab